shell/platform/android/io/flutter/plugin/common/MethodCall.java

```diff
@@ -0,0 +1,29 @@
+package io.flutter.plugin.common;
+
+import java.util.Objects;
+
+/**
+ * Command object representing a method call on a {@link FlutterChannel}.
+ */
+public final class MethodCall {
+    /**
+     * The name of the called method.
+     */
+    public final String method;
+
+    /**
+     * Arguments for the call.
+     */
+    public final Object arguments;
```

Comment:
> I'm not sure I understand what we expect to gain by having two channel types. Is it mainly to provide an API with more precise types? Or to reserve the ability to define alternative communication protocols similar to method calls and event streams without having to add methods to a single channel and codec concept?

The latter.  We already have three patterns (messages, method calls, and streams).  It seems likely we'll have more over time.  If we decouple things now, this mechanism will be easier to extend in the future.